agent.go

```diff
@@ -985,9 +987,30 @@ func (a *Agent) sendBindingSuccess(m *stun.Message, local, remote Candidate) {
 	}
 }
 
+// Removes pending binding requests that are over maxBindingRequestTimeout old
+func (a *Agent) invalidatePendingBindingRequests() int {
+	nRemoved := 0
+	temp := a.pendingBindingRequests[:0]
+	for _, bindingRequest := range a.pendingBindingRequests {
+		if time.Since(bindingRequest.timestamp) < maxBindingRequestTimeout {
+			temp = append(temp, bindingRequest)
+		} else {
+			nRemoved++
+		}
+	}
+	// Garbage collect removed elements
+	for i := len(temp); i < len(a.pendingBindingRequests); i++ {
+		a.pendingBindingRequests[i] = bindingRequest{}
+	}
+	a.pendingBindingRequests = temp
+	return nRemoved
+}
+
 // Assert that the passed TransactionID is in our pendingBindingRequests and returns the destination
 // If the bindingRequest was valid remove it from our pending cache
 func (a *Agent) handleInboundBindingSuccess(id [stun.TransactionIDSize]byte) (bool, *bindingRequest) {
+	a.invalidatePendingBindingRequests()
```

Comment:
`Agent.handleInboundBindingSuccess` now runs two for-loops over the same list, which is not ideal. The benefit is that the functionality for invalidating pending requests and identifying the valid request is separated. Let me know if you want to move the invalidation logic into the `handleInboundBindingSuccess` method.

Reply:
I think this is fine! We can measure and rewrite if it is a problem